platform/windows/detect.py

```diff
@@ -417,7 +417,7 @@ def spawn_capture(sh, escape, cmd, args, env):
             ret = old_spawn(sh, escape, cmd, args, env)
 
             try:
-                with open(tmp_stdout_name, encoding="oem", errors="replace") as tmp_stdout:
+                with open(tmp_stdout_name, "r", encoding="utf-8", errors="replace") as tmp_stdout:
```

Comment:
Does VS writes to console using `oem` encoding?

Reply:
> The encoding selected is that of python's stdout stream. However, the temporary stdout file is populated by the output of the msvc compiler and linker commands. SCons uses encoding "oem" in some cases when processing msvc command output.

From: #91883. Not sure, if there's a way to tell what encoding is used.

Reply:
`SCons` does use `oem` to read `MSVC` output, but it's in a different context: https://github.com/SCons/scons/blob/a6f498306730b333c81cef8bbf135e26e3acb360/SCons/Tool/MSCommon/common.py#L387-L406. While `Godot` creates a file in a console command, `SCons` doen't create any files and uses a ["pipe"](https://docs.python.org/3/library/subprocess.html#subprocess.PIPE) (have no idea what it is, honestly). So `sys.stdout.encoding` feels more correct